apps/gallery/js/thumbnail_item.js

```diff
@@ -32,25 +33,16 @@ function ThumbnailItem(fileData) {
   var url = URL.createObjectURL(fileData.metadata.thumbnail);
   this.imgNode.src = url;
   this.htmlNode.appendChild(this.imgNode);
-
-  this.localize();
 }
 
-ThumbnailItem.formatter = new Intl.DateTimeFormat(navigator.languages, {
-  hour: 'numeric',
-  minute: 'numeric',
-  month: 'long',
-  day: 'numeric',
-  year: 'numeric',
-});
-
 ThumbnailItem.prototype.localize = function() {
   var date = new Date(this.data.date);
+  var formatter = IntlHelper.get('time-stamp');
 
   var descId = !this.data.metadata.video ?
     'imageDated' : 'videoDated';
 
   navigator.mozL10n.setAttributes(this.imgNode, descId, {
-    timeStamp: ThumbnailItem.formatter.format(date)
+    timeStamp: formatter.format(date)
```

Comment:
it doesn't seem that you are doing `IntlHelper.observer('time-stamp')` anywhere. How will this code react to this intl object being modified? (locale change, timeformat change, moztimechange are all covered by IntlHelper.observe)